recipes/root/all/test_package/conanfile.py

```diff
@@ -0,0 +1,21 @@
+import os
+
+from conans import CMake, ConanFile, RunEnvironment, tools
+
+
+class RootTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = ("cmake_paths", "cmake_find_package")
+    requires = ("catch2/2.13.3",)
```

Comment:
The test packages are supposed to be more minimal to make sure the header(s) and link libs are present.

Given the size of this project, it hard to say but it looks like it's bit big

Reply:
I have added two tests:

(1) Fill a histogram with some random numbers.

This test is just to check that packages can find and use the ROOT libraries. It could be reduced eg to just call the constructor.

(2) Write a non-trivial ROOT file and check that it can be read back correctly.

This test is important. To read ROOT files, ROOT needs to be able to find "dictionary" files at run time. It is possible to end up in a situation where these files are included in the package directory but ROOT fails to find them at runtime. You will be able to successfully build an application but it will crash when you try to read a file.

The only way that I know to be sure that those files will actually be found at runtime is to actually try writing and reading some data.

It looks like on `Linux x86_64, Release, clang 8` one of these tests failed, albeit with an unhelpful error message, demonstrating the importance of these tests.

If the running time of these tests is a concern, I can reduce the amount of data written/read.

I could also remove the catch2 dependency (it's not clear from your message if this is a concern?). This is not essential and could be removed.

Reply:
> one of these tests failed, albeit with an unhelpful error message, demonstrating the importance of these tests.

That's what I was expecting. I think we'll leave it to your judgement above reducing it's size (if at all).

Catch is not the issues but the overall shape/size

Looking at it again, would it be possible to have everything in 1 cpp file? moving Event and Partical to the same `test_pacakge.cpp`

Reply:
I have tried to reduce the size of the test code as much as possible in 7f503ea.

> Looking at it again, would it be possible to have everything in 1 cpp file? moving Event and Partical to the same 
> test_pacakge.cpp

I have merged `Event.hpp/.cpp` and `Particle.hpp/.cpp` into a single `hpp/cpp` pair. 

At minimum, the `ROOT_GENERATE_DICTIONARIES` macro (which does some magic and generates whatever ROOT needs to read/write files) requires 1 header and 1 `Linkdef` file. I could reduce the file count by 1 by putting the code from the `cpp` into the header. But I don't think that would really help readability.

Reply:
I also removed test (1). Test (2) also requires that you can link to the ROOT libraries so test (1) was redundant.